pint/utils.py

```diff
@@ -370,11 +371,9 @@ def taylor_horner(x, coeffs):
     Out[1]: 40.0
 
     """
-    result = 0.0
-    fact = float(len(coeffs))
-    for coeff in coeffs[::-1]:
-        result = result * x / fact + coeff
-        fact -= 1.0
+    fact = factorial(range(len(coeffs)-1, -1, -1))
```

Comment:
Hmmm...  I'm not sure that I like changing the version of taylor_horner that I had to using factorial() and np.polyval()..  I think that this new version us not as stable floating-point-wise.  That's because for high-order expansions, when the factorial values become very large, you end up with large terms in the numerator and denominator  which can cause floating point issues.  With the older implementation, the factorial calculation is accumulated piecewise as x/fact which means that you never get huge numbers.  Also, a minor point is that np.flipud() is kind of opaque for 1-D arrays or lists.  you should use coeffs[::-1] which is Python for "a copy of the array but in reversed order".